codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointProviderSpec.java

```diff
@@ -17,191 +17,281 @@
 
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.CodeBlock;
-import com.squareup.javapoet.FieldSpec;
 import com.squareup.javapoet.MethodSpec;
-import com.squareup.javapoet.ParameterSpec;
-import com.squareup.javapoet.ParameterizedTypeName;
 import com.squareup.javapoet.TypeName;
 import com.squareup.javapoet.TypeSpec;
-import com.squareup.javapoet.TypeVariableName;
-import com.squareup.javapoet.WildcardTypeName;
-import java.net.URI;
-import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import javax.lang.model.element.Modifier;
 import software.amazon.awssdk.annotations.SdkInternalApi;
-import software.amazon.awssdk.awscore.endpoints.AwsEndpointAttribute;
 import software.amazon.awssdk.codegen.model.config.customization.EndpointAuthSchemeConfig;
+import software.amazon.awssdk.codegen.model.config.customization.KeyTypePair;
 import software.amazon.awssdk.codegen.model.intermediate.IntermediateModel;
 import software.amazon.awssdk.codegen.model.intermediate.Metadata;
-import software.amazon.awssdk.codegen.model.rules.endpoints.BuiltInParameter;
 import software.amazon.awssdk.codegen.model.rules.endpoints.ParameterModel;
+import software.amazon.awssdk.codegen.model.rules.endpoints.RuleModel;
+import software.amazon.awssdk.codegen.model.service.EndpointRuleSetModel;
 import software.amazon.awssdk.codegen.poet.ClassSpec;
 import software.amazon.awssdk.codegen.poet.PoetUtils;
+import software.amazon.awssdk.codegen.poet.rules.EndpointRulesSpecUtils;
 import software.amazon.awssdk.core.exception.SdkClientException;
-import software.amazon.awssdk.endpoints.Endpoint;
 import software.amazon.awssdk.utils.CompletableFutureUtils;
-import software.amazon.awssdk.utils.Logger;
 import software.amazon.awssdk.utils.Validate;
 
 public class EndpointProviderSpec implements ClassSpec {
```

Comment:
It looks like there are a lot of changes here, but this is actually just `EndpointProviderSpec2`, which has already been code reviewed and is listed as a deletion below. You can skip reviewing this class if you want, since it wasn't changed.

Reply:
Looks like it wasnt merged in ? Would be great if you could provide a PR link

Reply:
https://github.com/aws/aws-sdk-java-v2/blame/9a3c99e5b70f608d20c18248aa9bdb5c8d98b684/codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules2/EndpointProviderSpec2.java is the file already in the repo.

Original PR for it was https://github.com/aws/aws-sdk-java-v2/commit/db355d31b851a617b5d2c39b52b9fdcd7f2b7d4b

Reply:
got it thanks